web/src/app/profile/page.tsx

```diff
@@ -1,58 +1,179 @@
 'use client';
 
-import { useState, useEffect } from 'react';
-import { getUserInfoMeGet, UserOut } from '@/gen';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { getUserInfoMeGet, getUserStatisticsStatsGet } from '@/gen';
+import MetricCard from '@/components/MetricCard';
+import { ActivityHeatmap, HeatmapData } from '@/components/ActivityHeatmap';
 
 export default function Profile() {
-    const [userState, setUserState] = useState({
-        data: null as UserOut | null,
-        loading: true,
-        error: null as string | null,
+    const {
+        data: profile,
+        isLoading: profilePending,
+        isError: profileError,
+    } = useQuery({
+        queryKey: ['profile'],
+        queryFn: () => getUserInfoMeGet(),
     });
 
-    useEffect(() => {
-        fetchUser();
-    }, []);
-
-    async function fetchUser() {
-        try {
-            setUserState((prev) => ({ ...prev, error: null }));
-            const res = await getUserInfoMeGet();
-            setUserState((prev) => ({
-                ...prev,
-                data: res.data || null,
-                loading: false,
-            }));
-        } catch (err: any) {
-            setUserState({
-                data: null,
-                loading: false,
-                error:
-                    err.detail ?? 'There was an error while fetching profile',
-            });
+    const {
+        data: stats,
+        isLoading: statsPending,
+        isError: statsError,
+    } = useQuery({
+        queryKey: ['stats'],
+        queryFn: () => getUserStatisticsStatsGet(),
+    });
+
+    const queryClient = useQueryClient();
+
+    function formatHeatmapData(reviews_by_date: { [key: string]: number }) {
+        const heatmapData: HeatmapData = {};
+
+        for (let date in reviews_by_date) {
+            const nrReviews = reviews_by_date[date];
+            heatmapData[date] = {
+                date: date,
+                nrReviews: nrReviews,
+                intensity:
+                    nrReviews === 0
+                        ? 0
+                        : Math.min(Math.floor(nrReviews / 10) + 1, 4),
```

Comment:
Some users may end up doing 100s of reviews per day, so we should probably have more and/or bigger intensity steps here later on.
EDIT: or maybe not, let's see later on.